loaders/dashboards/pkg/util/grafana_util.go

```diff
@@ -61,14 +61,16 @@ func SetRequest(method string, url string, body io.Reader, retry int) ([]byte, i
 		resp, err = getHTTPClient().Do(req)
 	}
 
-	if resp != nil {
-		defer resp.Body.Close()
-		respBody, err := io.ReadAll(resp.Body)
-		if err != nil {
-			klog.Info("failed to parse response body ", "error ", err)
-		}
-		return respBody, resp.StatusCode
-	} else {
+	if resp == nil {
 		return nil, http.StatusNotFound
 	}
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
```

Comment:
I think we can improve this to defer the body closure and also avoid the linting for not handling the error:

```go
defer func() {
  err := resp.Body.Close()
  if err != nil {
    // log the error?
  }
}()
```

This applies to all the `req.Body.Close()` statements.

Reply:
using this pattern now.